proto/penumbra/penumbra/core/component/dex/v1/dex.proto

```diff
@@ -651,6 +651,12 @@ message EventPositionExecution {
   num.v1.Amount reserves_1 = 3;
   // The reserves of asset 2 of the position after execution.
   num.v1.Amount reserves_2 = 4;
+  // The reserves of asset 1 of the position before execution.
+  num.v1.Amount prev_reserves_1 = 5;
+  // The reserves of asset 2 of the position before execution.
+  num.v1.Amount prev_reserves_2 = 6;
+  // Context: the end-to-end route that was being traversed during execution.
+  DirectedTradingPair context = 7;
```

Comment:
I made this a single-field `DirectedTradingPair` rather than two fields as in discord with the thought that it would make it easier to make queries like "show me all the position executions that occurred in the context of the swap X=>Y at height ABC"